### PR TITLE
fix(tasks): Enforce consistent application task ordering

### PR DESCRIPTION
**Pull request by robzienert**
There have been some odd reports that, in some cases while using the `SqlExecutionRepository`, some applications' tasks are unsorted. This is weird, because SQL itself orders all executions that are returned from the repository. It turns out that this ordering is lost and sort of mangled in the TaskController.

This change moves the sorting back into the ExecutionRepository and forces canned comparators which can be implemented appropriately by the implementing backend. For Redis, this will continue to use the old runtime sorting strategy, whereas SQL will use SQL.

Removed the `RedisPollingAgentExecutionRepositorySpec` class as it doesn't actually test anything `RedisExecutionRepositorySpec` doesn't.

### CODE REVIEW

File: orca-redis/src/main/java/com/netflix/spinnaker/orca/pipeline/persistence/jedis/RedisExecutionRepository.java [hunk 1]
```diff
@@ -464,8 +464,8 @@ public void delete(@Nonnull ExecutionType type, @Nonnull String id) {
   }
 
   @Override
-  public @Nonnull
-  Observable<Execution> retrieveOrchestrationsForApplication(@Nonnull String application, @Nonnull ExecutionCriteria criteria) {
+  public @Nonnull Observable<Execution> retrieveOrchestrationsForApplication(@Nonnull String application,
+                                                                             @Nonnull ExecutionCriteria criteria) {
```
Comment (emjburns) [inline orca-redis/src/main/java/com/netflix/spinnaker/orca/pipeline/persistence/jedis/RedisExecutionRepository.java]: Is this your preferred way to do multi line function declarations with long names?

File: orca-redis/src/main/java/com/netflix/spinnaker/orca/pipeline/persistence/jedis/RedisExecutionRepository.java [hunk 2]
```diff
@@ -545,6 +545,32 @@ public void delete(@Nonnull ExecutionType type, @Nonnull String id) {
     return currentObservable;
   }
 
+  @Nonnull
+  @Override
+  public List<Execution> retrieveOrchestrationsForApplication(@Nonnull String application,
+                                                              @Nonnull ExecutionCriteria criteria,
+                                                              @Nullable ExecutionComparator sorter) {
+
```
Comment (emjburns) [inline orca-redis/src/main/java/com/netflix/spinnaker/orca/pipeline/persistence/jedis/RedisExecutionRepository.java]: extra line

**Comment by robzienert** [comment]
Now with passing tests!

File: orca-redis/src/main/java/com/netflix/spinnaker/orca/pipeline/persistence/jedis/RedisExecutionRepository.java (hunk 1 above)
Comment (robzienert) [reply orca-redis/src/main/java/com/netflix/spinnaker/orca/pipeline/persistence/jedis/RedisExecutionRepository.java]: This is consistent with most of the Java/Groovy code in our projects. Kotlin has a different standard, which I'm fine with (new lining the first argument and indenting one). The key thing, for me, is not going over 120 characters where feasible.

I really dislike side-scrolling. :)

**Pull request merged**